src/components/MultiHopTrade/components/TradeConfirm/hooks/useTradeButtonProps.tsx

```diff
@@ -0,0 +1,159 @@
+import { useCallback, useMemo, useState } from 'react'
+import { useHistory } from 'react-router-dom'
+import { useGetTradeQuotes } from 'components/MultiHopTrade/hooks/useGetTradeQuotes/useGetTradeQuotes'
+import { TradeRoutePaths } from 'components/MultiHopTrade/types'
+import { assertUnreachable } from 'lib/utils'
+import {
+  selectActiveQuote,
+  selectConfirmedTradeExecutionState,
+  selectHopExecutionMetadata,
+} from 'state/slices/tradeQuoteSlice/selectors'
+import { tradeQuoteSlice } from 'state/slices/tradeQuoteSlice/tradeQuoteSlice'
+import {
+  HopExecutionState,
+  TradeExecutionState,
+  TransactionExecutionState,
+} from 'state/slices/tradeQuoteSlice/types'
+import { useAppDispatch, useAppSelector, useSelectorWithArgs } from 'state/store'
+
+import { useTradeExecution } from '../../MultiHopTradeConfirm/hooks/useTradeExecution'
+import { getHopExecutionStateButtonTranslation } from '../helpers'
+import { useActiveTradeAllowance } from './useSignAllowanceApproval'
+
+type UseTradeButtonPropsProps = {
+  tradeQuoteStep: TradeQuoteStep
+  currentHopIndex: SupportedTradeQuoteStepIndex
+  activeTradeId: string
+  isExactAllowance: boolean
+}
+
+type TradeButtonProps = {
+  handleSubmit: () => void
+  buttonText: string
+  isLoading: boolean
+  isDisabled: boolean
+}
+
+export const useTradeButtonProps = ({
+  tradeQuoteStep,
+  currentHopIndex,
+  activeTradeId,
+  isExactAllowance,
+}: UseTradeButtonPropsProps): TradeButtonProps | undefined => {
+  const [isSignTxLoading, setIsSignTxLoading] = useState(false)
+  const dispatch = useAppDispatch()
+  const history = useHistory()
+  const confirmedTradeExecutionState = useAppSelector(selectConfirmedTradeExecutionState)
+  const activeQuote = useAppSelector(selectActiveQuote)
+  const { isFetching, data: tradeQuoteQueryData } = useGetTradeQuotes()
+  const {
+    handleSignAllowanceApproval,
+    isAllowanceApprovalLoading,
+    isAllowanceApprovalPending,
+    handleSignAllowanceReset,
+    isAllowanceResetLoading,
+    isAllowanceResetPending,
+    handleSignPermit2,
+  } = useActiveTradeAllowance({
+    tradeQuoteStep,
+    isExactAllowance,
+    activeTradeId,
+  })
+  const handleTradeConfirm = useCallback(() => {
+    if (!activeQuote) return
+    dispatch(tradeQuoteSlice.actions.confirmTrade(activeQuote.id))
+  }, [dispatch, activeQuote])
+
+  const hopExecutionMetadataFilter = useMemo(() => {
+    return {
+      tradeId: activeTradeId ?? '',
+      hopIndex: currentHopIndex ?? 0,
+    }
+  }, [activeTradeId, currentHopIndex])
+  const {
+    state: hopExecutionState,
+    swap: { state: swapTxState },
+  } = useSelectorWithArgs(selectHopExecutionMetadata, hopExecutionMetadataFilter)
+
+  const handleTradeConfirmSubmit = useCallback(() => {
+    // TODO: Implement me
+    // if (isModeratePriceImpact) {
+    //   setShouldShowWarningAcknowledgement(true)
+    // } else {
+    //   handleTradeConfirm()
+    // }
```

Comment:
👀

Reply:
Intentional, just need to uncomment and implement (in this PR if I have time before you review tonight, or in another one).

Reply:
Resolved in https://github.com/shapeshift/web/pull/8192/commits/cdb3d563fa54e61c1aaa153c70c4e04043bd928e (implemented the warning ack).